proxy.go

```diff
@@ -241,14 +241,22 @@ func executeWithRetry(
 			// comment s.host.dec() line to avoid double increment; issue #322
 			// s.host.dec()
 			atomic.StoreUint32(&s.host.active, uint32(0))
-			newHost := s.host.replica.cluster.getHost()
+			nextHost := s.host.replica.cluster.getHost()
 			// The query could be retried if it has no stickiness to a certain server
-			if numRetry < maxRetry && newHost.isActive() && s.sessionId == "" {
+			if numRetry < maxRetry && nextHost.isActive() && s.sessionId == "" {
 				// the query execution has been failed
 				monitorRetryRequestInc(s.labels)
-
+				currentHost := s.host
+
+				// decrement the current failed host counter and increment the new host
```

Comment:
It might also a be a good idea to link the PR.

Reply:
done